Result/Result.swift

```diff
@@ -179,6 +179,15 @@ public func ?? <T, Error> (left: Result<T, Error>, @autoclosure right: () -> Res
 	return left.recoverWith(right())
 }
 
+// MARK: - Derive result from failable closure
+
+public func materialize<T>(f: () throws -> T) -> Result<T, ErrorType> {
```

Comment:
My concern is `materialize` function can't be called without type annotation.

If error type can't be documented in throws clause, there is no place to specify `E` except call-site. That is, type of `let result` can't be inferred from return value of `materialized` function. Consequently, type annotation (`Result<AnyObject, NSError>` in my code above) is required to call `materialized`.